effects/push-cache/default.nix

```diff
@@ -0,0 +1,160 @@
+{ lib, withSystem, config, ... }:
+let pkgs-x86_64-linux = withSystem "x86_64-linux" ({ pkgs, ... }: pkgs);
```

Comment:
We need to support other systems.

Reply:
I would see what happens if we try to push non `x86_64-linux` derivations using an `x86_64-linux` effect before.

Reply:
Those will be scheduled during the build phase of the job, and substituted into the agent's store before the effect is launched.

Reply:
Tested: pushing derivations built on other systems to attic works.